### PR TITLE
adding remaining study proposal form fields and style

### PR DESCRIPTION
**Pull request by jbwilson8**
[OTTER-46](https://openstax.atlassian.net/browse/OTTER-46)

**TODO:**  make button functionality better

[OTTER-46]: https://openstax.atlassian.net/browse/OTTER-46?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/researcher/study/[encodedStudyId]/edit/form.tsx [hunk 1]
```diff
@@ -1,84 +1,130 @@
...
+import Link from 'next/link'
 
 export const Form: React.FC<{ studyId: string; study: FormValues }> = ({ studyId, study }) => {
-    const { control } = useForm<FormValues>({
+    const {
+        control,
+        // formState: { isValid },
+        getValues,
+    } = useForm<FormValues>({
         resolver: zodResolver(schema),
```
Comment (chrisbendel) [inline src/app/researcher/study/[encodedStudyId]/edit/form.tsx:19]: This is a nitpick but i prefer having the schema defined  in the same file (its just easier for me to read) but no changes here necessary if your team prefers otherwise

**Review by chrisbendel — COMMENTED** [review]
I was going down a bit of a rabbit hole with the form stuff. A couple small comments, but maybe a larger discussion - is there a reason why @safeinsights/otter-devs decided to bring in `react-hook-form`? It looks like y'all are using some [adapter library](https://github.com/aranlucas/react-hook-form-mantine) which their last update was ~1 year ago, im not sure if it's still maintained. 

If you want to use react-hook-form with mantine components you might have to wrap all of the mantine input components with a `<Controller>` which gets super cumbersome, or it leads you to abstract those out with a library like the one you're using which is more maintenance. 

TLDR is there any reason why we can't just use the mantine `useForm` hook? i understand it might not provide every feature that react-hook-form does but its one less dependency and it also allows integration with zod/yup/etc

Hope im not derailing this PR with this, but thought it might be important to bring up for cross team learning and alignment!

File: src/app/researcher/study/[encodedStudyId]/edit/schema.ts [hunk 2]
```diff
@@ -2,14 +2,21 @@ import { z } from 'zod'
...
+        piName: z.string().min(3).max(100),
+        highlights: z.preprocess((value) => value === 'on', z.boolean()).nullish(),
+        eventCapture: z.preprocess((value) => value === 'on', z.boolean()).nullish(),
+        outputMimeType: z.string().nullish(),
+        irbDocument: z.string().nullish(),
+        containerLocation: z.string().min(3).max(250),
+    })
+    .refine((data) => data.highlights || data.eventCapture, {
+        message: 'At least one checkbox must be selected',
+        path: ['highlights', 'evntCapture'],
```
Comment (chrisbendel) [inline src/app/researcher/study/[encodedStudyId]/edit/schema.ts:18]: Spelling error on 'evntCapture' which might have been leading you to manually check the valid state?

File: src/app/researcher/study/[encodedStudyId]/edit/form.tsx [hunk 3]
```diff
@@ -1,84 +1,130 @@
...
         defaultValues: study,
+        mode: 'onChange',
     })
+    const [isDisabled, setIsDisabled] = useState(true)
+    const [isFormValid, setIsFormValid] = useState(true)
+    const enableButton = () => {
+        setIsDisabled(false)
+    }
+    const checkFormValid = () => {
```
Comment (chrisbendel) [inline src/app/researcher/study/[encodedStudyId]/edit/form.tsx:28]: I started digging into "why" you might be needing this - could be related to a spelling error in `schema.ts` in the `refine` callback

File: src/app/researcher/study/[encodedStudyId]/edit/schema.ts (hunk 2 above)
Comment (jbwilson8) [reply src/app/researcher/study/[encodedStudyId]/edit/schema.ts:18]: Thanks for catching that

**Comment by nathanstitt** [comment]
Hey @chrisbendel thanks for taking a look at the form stuff ❤️     

We choose react-hook-form over mantine forms because it's more efficient, matine forms re-renders the entire form on every event while react-hook-form only re-renders the field that's being interacted with.  [comment to that effect](https://github.com/orgs/mantinedev/discussions/5263) but I also verified it. That's not a huge deal but react-hook-form is also very popular and I suspected docs would be more available.

Use of the adapter did make me a little nervous but each component is just a tiny wrapper so it seems like it should be bug-free https://github.com/aranlucas/react-hook-form-mantine/blob/master/src/Input/Input.tsx for an example.

File: src/app/researcher/study/[encodedStudyId]/edit/form.tsx (hunk 1 above)
Comment (nathanstitt) [reply src/app/researcher/study/[encodedStudyId]/edit/form.tsx:19]: putting the schema in a separate file lets us re-use it for server-side validation in the `actions.ts` file to validate the incoming values before using them with the DB.    It also can generate the typescript types for the form and that can be imported separately as well

**Comment by chrisbendel** [comment]
> Hey @chrisbendel thanks for taking a look at the form stuff ❤️
> 
> We choose react-hook-form over mantine forms because it's more efficient, matine forms re-renders the entire form on every event while react-hook-form only re-renders the field that's being interacted with. [comment to that effect](https://github.com/orgs/mantinedev/discussions/5263) but I also verified it. That's not a huge deal but react-hook-form is also very popular and I suspected docs would be more available.
> 
> Use of the adapter did make me a little nervous but each component is just a tiny wrapper so it seems like it should be bug-free https://github.com/aranlucas/react-hook-form-mantine/blob/master/src/Input/Input.tsx for an example.

That makes sense! Just for y'alls knowledge it looks like a more recent version of the mantine/forms library has an [uncontrolled mode](https://mantine.dev/form/uncontrolled/) which [does not re-render all the components](https://github.com/orgs/mantinedev/discussions/1193#discussioncomment-9883362)
